app/modules/auth/auth_router.py

```
@@ -27,10 +45,229 @@
+            )
+        except Exception as e:
+            logger.error(
+                f"Error linking accounts after duplicate email error: {str(e)}"
+            )
+            db.rollback()
+            return None
+
+    @staticmethod
+    def _create_provider_info(
+        provider_data: list, oauth_token: str, uid: str, email: str
+    ) -> Dict[str, Any]:
```

Ruff (0.14.8)

Undefined name `Dict`

(F821)